crates/router/src/core/payments.rs

```diff
@@ -3685,14 +3692,57 @@ pub async fn decide_multiplex_connector_for_normal_or_recurring_payment<F: Clone
                     .as_ref()
                     .ok_or(errors::ApiErrorResponse::InternalServerError)
                     .attach_printable("Failed to find the merchant connector id")?;
+                if is_network_token_with_transaction_id_flow(
+                    is_connector_agnostic_mit_enabled,
+                    is_network_tokenization_enabled,
+                    payment_method_info,
+                ) {
+                    if let Ok((token_exp_month, token_exp_year)) =
+                        do_status_check_for_network_token(state, key_store, payment_method_info)
+                            .await
+                    {
+                        logger::info!(
+                            "using network_tokenization with transaction_id for MIT flow"
+                        );
+
+                        let network_transaction_id = payment_method_info
+                            .network_transaction_id
+                            .as_ref()
+                            .ok_or(errors::ApiErrorResponse::InternalServerError)?;
+
+                        let mandate_reference_id =
+                            Some(payments_api::MandateReferenceId::NetworkTokenWithNTI(
+                                payments_api::NetworkTokenWithNTIRef {
+                                    network_transaction_id: network_transaction_id.to_string(),
+                                    token_exp_month,
+                                    token_exp_year,
+                                },
+                            ));
 
-                if is_network_transaction_id_flow(
+                        connector_choice = Some((connector_data, mandate_reference_id.clone()));
+                        break;
+                    } else {
+                        logger::info!("using network_transaction_id for MIT flow"); //
```

Comment:
This log line should be different from the one in `is_network_transaction_id_flow` so that we get to know that we switch from the network tokenization flow to network transaction id flow